doc/ensemble.rst

```diff
@@ -0,0 +1,739 @@
+There are **three** Ensemble configuration that users can adjust to customize
+Ensemble behavior:
+
+1. :ref:`Manually<append_init>`: Create each Model individually and subsequently append the simulation to the ensemble.
+2. :ref:`Replica creation<replicas_init>`: Generate a specified number of copies or instances of a simulation.
+3. :ref:`Parameter expansion<param_expansion_init>`: Explore a range of simulation variable assignments by
+   specifying values of input parameters and selecting an expansion strategy to rearrange
+   across multiple Models within the ensemble.
+
+==============
+Initialization
+==============
+--------
+Overview
+--------
+The creation of an Ensemble involves the use of various initializer arguments,
+each serving a specific role in defining the characteristics and behavior of the
+ensemble. These arguments facilitate three distinct methods of ensemble
+creation: **manual model appending**, **parameter expansion**, and the utilization of **replicas**.
+
+The key initializer arguments are:
+
+-  `name` (Type: str): Specify the name of the ensemble, aiding in its unique identification.
+-  `params` (Type: dict[str, Any]): Provides a dictionary of parameters:values for expanding into the ``Model`` members within the ensemble. Enables parameter expansion for diverse scenario exploration.
+-  `params_as_args` (Type: list[str]): Specify which parameters from the `params` dictionary should be treated as command line arguments when executing the Models.
+-  `batch_settings` (Type: BatchSettings, optional): Describes settings for batch workload treatment.
+-  `run_settings` (Type: RunSettings, optional): Describes execution settings for individual Model members.
+-  `replicas` (Type: int, optional): Declare the number of ``Model`` clones within the ensemble, crucial for the creation of simulation replicas.
+-  `perm_strategy` (Type: str): Specifies a strategy for parameter expansion into ``Model`` instances, influencing the method of ensemble creation and number of ensemble members.
+    The options are `"all_perm"`, `"step"`, and `"random"`.
+
+-------------------
+Parameter Expansion
+-------------------
+.. _param_expansion_init:
+In ensemble simulations, parameter expansion is a technique that
+allows users to set parameter values using the `params` initializer.
+User's may control how the parameter values spread across the ensemble
+members by using the `perm_strategy` initializer.
+
+**Parameter Expansion Strategies:**
+
+-  `"all_perm"`: Generates all possible parameter permutations for an exhaustive exploration.
+-  `"step"`: Creates sets for each element in n arrays, providing a systematic exploration.
+-  `"random"`: Allows random selection from predefined parameter spaces, offering a stochastic approach.
+
+Examples
+--------
+Example 1 : Parameter Expansion with ``RunSettings`` and `params`
+    .. note::
+        The Ensemble requires an allocation.
```

Comment:
I'm not sure the ensemble requires an allocation in all cases. A local launcher should execute ok?

![image](https://github.com/CrayLabs/SmartSim/assets/3595025/979aadf1-86d1-4c87-9e8c-6c2b45d7a5c0)